packages/state-transition/src/metrics.ts

```diff
@@ -1,30 +1,111 @@
-import {Epoch} from "@lodestar/types";
-import {Gauge, Histogram} from "@lodestar/utils";
+import {MetricsRegister} from "@lodestar/utils";
 import {EpochTransitionStep} from "./epoch/index.js";
 import {StateCloneSource, StateHashTreeRootSource} from "./stateTransition.js";
 import {CachedBeaconStateAllForks} from "./types.js";
 
-export type BeaconStateTransitionMetrics = {
-  epochTransitionTime: Histogram;
-  epochTransitionCommitTime: Histogram;
-  epochTransitionStepTime: Histogram<{step: EpochTransitionStep}>;
-  processBlockTime: Histogram;
-  processBlockCommitTime: Histogram;
-  stateHashTreeRootTime: Histogram<{source: StateHashTreeRootSource}>;
-  numEffectiveBalanceUpdates: Gauge;
-  preStateBalancesNodesPopulatedMiss: Gauge<{source: StateCloneSource}>;
-  preStateBalancesNodesPopulatedHit: Gauge<{source: StateCloneSource}>;
-  preStateValidatorsNodesPopulatedMiss: Gauge<{source: StateCloneSource}>;
-  preStateValidatorsNodesPopulatedHit: Gauge<{source: StateCloneSource}>;
-  preStateClonedCount: Histogram;
-  postStateBalancesNodesPopulatedMiss: Gauge;
-  postStateBalancesNodesPopulatedHit: Gauge;
-  postStateValidatorsNodesPopulatedMiss: Gauge;
-  postStateValidatorsNodesPopulatedHit: Gauge;
-  newSeenAttestersPerBlock: Gauge;
-  newSeenAttestersEffectiveBalancePerBlock: Gauge;
-  attestationsPerBlock: Gauge;
-};
+export type BeaconStateTransitionMetrics = ReturnType<typeof getMetrics>;
+
+/**
+ * A collection of metrics used throughout the State Transition.
+ */
+export function getMetrics(register: MetricsRegister) {
+  // Using function style instead of class to prevent having to re-declare all MetricsPrometheus types.
+
+  return {
+    epochTransitionTime: register.histogram({
```

Comment:
it's nice to move metric definition to state-transition but I'd like to add a label to this
when we run stateTransition from normal flow, pass in `BlockSource`
when we run from regen, pass in `regen`
when we run from historical state regen, pass in `historical_regen`

and deduplicate the same set of metrics there, then update dashboard with the label, that would make the work more valuable for us

Reply:
> and deduplicate the same set of metrics there, then update dashboard with the label, that would make the work more valuable for us

that's good idea, we could address this in a follow-up PR

Reply:
sounds good to me